morpho5/geometry/mesh.c

```diff
@@ -224,7 +224,7 @@ bool mesh_setconnectivityelement(objectmesh *mesh, unsigned int row, unsigned in
     unsigned int indx[2]={row,col};
     if (mesh_checkconnectivity(mesh)) {
         value old = MORPHO_NIL;
-        if (array_getelement(mesh->conn, 2, indx, &old) &&
+        if ((array_getelement(mesh->conn, 2, indx, &old)==ARRAY_OK) &&
```

Comment:
Great catch! The little test I had runs with a lot less memory now (see output of massif heap profiler below, before->left and after->right), and is also much faster!
![massif](https://user-images.githubusercontent.com/16670570/202875647-d2f352d6-d3b3-47df-9247-5668c9a59da9.png)